packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -189,7 +189,9 @@ abstract class ResidentWebRunner extends ResidentRunner {
     globals.printStatus('');
     globals.printStatus(message);
     const String quitMessage = 'To quit, press "q".';
-    globals.printStatus('For a more detailed help message, press "h". $quitMessage');
+    if (device.device is! WebServerDevice) {
+      globals.printStatus('For a more detailed help message, press "h". $quitMessage');
```

Comment:
It might be worth keeping the "h", except that the help message could explain how the server mode is different from the browser mode.

Reply:
I did add more details in the web server startup below (see web_devices.dart), that way it shows up in IDEs too